desktop-jamepad/src/main/java/de/golfgl/gdx/controllers/jamepad/support/JamepadController.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    @Override
+    public Vector3 getAccelerometer(int accelerometerCode) {
+        throw notSupported("getAccelerometer(...)");
+    }
+
+    @Override
+    public void setAccelerometerSensitivity(float sensitivity) {
+        throw notSupported("setAccelerometerSensitivity(...)");
+    }
+
+    @Override
+    public String getName() {
+        return query(new ControllerQuerier<String>() {
+            public String query(ControllerIndex controllerIndex) throws ControllerUnpluggedException {
+                return controllerIndex.getName();
+            }
+
+            public String valueOnFailure(ControllerIndex controllerIndex) {
+                return "Unknown";
+            }
+        });
+    }
+
+    @Override
+    public void addListener(ControllerListener listener) {
+        compositeControllerListener.addListener(listener);
+    }
+
+    @Override
+    public void removeListener(ControllerListener listener) {
+        compositeControllerListener.removeListener(listener);
+    }
+
+    public void update() {
+        updateButtonsState();
+        updateAxisState();
+    }
+
+    private ControllerButton toButton(int buttonCode) {
+        return CODE_TO_BUTTON.get(buttonCode);
+    }
+
+    private ControllerAxis toAxis(int axisCode) {
+        return CODE_TO_AXIS.get(axisCode);
+    }
+
+    private void updateAxisState() {
+        for (ControllerAxis axis : ControllerAxis.values()) {
+            int id = axis.ordinal();
+
+            float value = getAxis(id);
+            if (value != axisState.get(id)) {
+                if (logger.getLevel() == Logger.DEBUG) {
+                    logger.debug("Axis [" + id + " - " + toAxis(id) + "] moved [" + value + "]");
+                }
+                compositeControllerListener.axisMoved(this, id, value);
+            }
+            axisState.put(id, value);
+        }
+    }
+
+    private void updateButtonsState() {
+        for (ControllerButton button : ControllerButton.values()) {
+            int id = button.ordinal();
+
+            boolean pressed = getButton(id);
+            if (pressed != buttonState.get(id)) {
+                if (pressed) {
+                    compositeControllerListener.buttonDown(this, id);
+                } else {
+                    compositeControllerListener.buttonUp(this, id);
+                }
+
+                if (logger.getLevel() == Logger.DEBUG) {
+                    logger.debug("Button [" + id + " - " + toButton(id) + "] is " + (pressed ? "pressed" : "released"));
+                }
+            }
+            buttonState.put(id, pressed);
+        }
+    }
+
+    private <R> R query(ControllerQuerier<R> querier) {
+        for (int i = 0; i < MAX_QUERY_ATTEMPTS; i++) {
+            try {
+                return querier.query(controllerIndex);
+            } catch (ControllerUnpluggedException e) {
+                logger.error("Failed querying controller at index: " + controllerIndex.getIndex(), e);
```

Comment:
This line causes a lot of redundant information written to the error log when a controller is disconnected. Should better be just an logger.info without all the stacktrace stuff.

Reply:
The excessive logging smells like a different problem, meaning, it sounds like when the controller is disconnected we are not properly disposing of the controller and probably should not be querying the controller if it has been disconnected. Does that make sense?

Reply:
I think the querying is just done before the controller disconnect callback is done. The game crashes on the disconnect callback which comes half a second after the controller is disconnected. Why do you call reconnectController()?

Reply:
The hope was potentially reconnecting the controller and being able to continue. If you think that is overkill, I am cool with that going away.